lib/source/consul.js

```diff
@@ -1,358 +1,91 @@
-      const diff = deepDiff(this.properties.consul[name], this._healthWatchers[name].parser.properties);
-
-      if (diff) {
-        Log.log('SILLY', `Calculated ${this.name} properties diff during update`, {sourceName: 'diff', diff});
-      }
-
-      this.properties.consul[name] = this._healthWatchers[name].parser.properties;
-    } else {
-      // Empty data means the service has been deregistered.
-      this._unregisterHealthWatcher(name);
-      delete this.properties.consul[name];
-    }
-
-    this._okay = true;
-    this._updated = new Date();
-
-    this.emit('update', this, {
-      serviceName: name,
-      clusterName: cluster
-    });
-  }
-
-  /**
-   * Checks if there's already a watcher for the Consul health/service API
-   *
-   * @param {String}    name The name of the watcher to check
-   * @return {Boolean}  true if there's a watcher with the name; false otherwise
-   * @private
-   */
-  _hasHealthWatcher(name) {
-    return this._healthWatchers && this._healthWatchers[name] && this._healthWatchers[name].watcher;
-  }
-
-  /**
-   * Register a watcher that's tracking a Consul health/service API
-   * @param {String} name           The name of the watcher
-   * @param {Consul.Watch} watcher  The watcher returned by `consul.watch()`
-   * @private
-   */
-  _registerHealthWatcher(name, watcher) {
-    if (!this._healthWatchers) {
-      this._healthWatchers = {};
-    }
-    this._healthWatchers[name] = {
-      watcher,
-      parser: new ConquesoCompatHealthParser()
-    };
-  }
-
-  /**
-   * Unregister a watcher that's tracking a Consul health/service API
-   * @param {String} name  The name of the watcher
-   * @private
-   */
-  _unregisterHealthWatcher(name) {
-    if (this._hasHealthWatcher(name)) {
-      this._healthWatchers[name].watcher.end();
-      delete this._healthWatchers[name];
-    }
-  }
+  shutdown() {
+    super.shutdown();
 
-  /**
-   * Shut down everything tracking the Consul health/service API.
-   * @private
-   */
-  _shutdownHealthWatchers() {
-    /* eslint-disable guard-for-in */
-    // There's already a guard in Consul#_unregisterHealthWatchers
-    for (const name in this._healthWatchers) {
-      this._unregisterHealthWatcher(name);
+    if (this.watcher) {
+      this.watcher.end();
+      delete this.watcher;
     }
 
-    /* eslint-enable guard-for-in */
-    delete this._healthWatchers;
+    return this;
   }
 
   /**
-   * Shut down everything tracking the Consul service/list API.
-   * @private
-   */
-  _shutdownServiceWatcher() {
-    if (this._serviceWatcher) {
-      this._serviceWatcher.end();
-    }
-    delete this._serviceWatcher;
-  }
-
-  /**
-   * Generic handler for all errors
+   * Update the catalog cache before parsing results
    *
-   * @param {Error} err
-   * @private
+   * @param  {Object} data Payload from watcher event
    */
-  _error(err) {
-    this._okay = false;
+  _update(data) {
+    this.client.catalog.node.list((err, nodes) => {
```

Comment:
That's a good question. I'm leaning towards no because we're already cache all of the data from Consul in the parser, Rather than propagate flap during an unavailability event on the Consul cluster, we should just wait it out with the existing data.